service.go

```diff
@@ -152,16 +93,64 @@ func (s *service) GetAll(ctx context.Context) ([]InternalWebhook, error) {
 	return iws, nil
 }
 
-func prepArgusListenerClientConfig(cfg *ListenerConfig, watches ...Watch) {
-	logger := cfg.Logger
-	watches = append(watches, webhookListSizeWatch(cfg.Measures.WebhookListSizeGaugeName))
-	cfg.Config.Listener = chrysom.ListenerFunc(func(items chrysom.Items) {
+func NewService(client *chrysom.BasicClient) *service {
+	return &service{
+		argus: client,
+		now:   time.Now,
+	}
+}
+
+type ClientServiceIn struct {
+	fx.In
+
+	BasicClient *chrysom.BasicClient
+}
+
+// ProvideService builds the Argus client service from the given configuration.
+func ProvideService(in ClientServiceIn) Service {
+	return NewService(in.BasicClient)
+}
+
+// TODO: Refactor and move Watch and Listener related code to chrysom.
+type DefaultListenersIn struct {
+	fx.In
+
+	WebhookListSizeGauge prometheus.Gauge `name:"webhook_list_size"`
+}
+
+type DefaultListenerOut struct {
+	fx.Out
+
+	Watchers []Watch `group:"watchers,flatten"`
+}
+
+func ProvideDefaultListeners(in DefaultListenersIn) DefaultListenerOut {
+	var watchers []Watch
+
+	watchers = append(watchers, webhookListSizeWatch(in.WebhookListSizeGauge))
+
+	return DefaultListenerOut{
+		Watchers: watchers,
+	}
+}
+
+type ListenerIn struct {
+	fx.In
+
+	Shutdowner fx.Shutdowner
+	Watchers   []Watch `group:"watchers"`
+}
+
+func ProvideListener(in ListenerIn) chrysom.Listener {
```

Comment:
Startup logic for listener watchers (including the watcher interface) will be moved to `chrysom/listenerClient.go`, but it'll required additional old ancla code to be refactored